src/identity/mod.rs

```diff
@@ -0,0 +1,281 @@
+/// identities
+/// Different from `Id` as it is meant to be used to store only public key. So, only used to
+/// represent identity of other users
+pub struct PublicId {
+    /// RSA public key defining the user
+    rsa: Rsa<Public>,
+}
+
+impl Id {
+    /// Generate a new identity
+    /// # Errors
+    /// * [`AetherError::OpenSSLError`]   -   If the RSA key pair could not be generated
+    pub fn new() -> Result<Id, AetherError> {
+        Ok(Id {
+            rsa: Rsa::generate(RSA_SIZE)?,
+        })
+    }
+
+    /// Returns [`PathBuf`] to the private key on the filesystem
+    pub fn get_private_key_path() -> PathBuf {
+        let mut config = Self::get_config_dir();
+        config.push("private_key.pem");
+        config
+    }
+
+    /// Returns [`PathBuf`] to the public key on the filesystem
+    pub fn get_public_key_path() -> PathBuf {
+        let mut config = Self::get_config_dir();
+        config.push("public_key.pem");
+        config
+    }
+
+    /// Returns [`PathBuf`] to the config directory on the filesystem
+    fn get_config_dir() -> PathBuf {
+        match home_dir() {
+            Some(mut home) => {
+                home.push(".config/aether/");
+                match fs::create_dir_all(home.clone()) {
+                    Ok(()) => home,
+                    Err(_) => PathBuf::from("./"),
+                }
+            }
+            None => PathBuf::from("./"),
+        }
+    }
+
+    /// Save the current identity on the filesystem
+    /// Saves the public key and the private key in PEM format
+    pub fn save(&self) -> Result<(), AetherError> {
+        let rsa_public = self.rsa.public_key_to_pem()?;
+        let rsa_private = self.rsa.private_key_to_pem()?;
+
+        if let Err(err) = fs::write(Self::get_private_key_path(), rsa_private) {
+            Err(AetherError::FileWrite(err))
+        } else if let Err(err) = fs::write(Self::get_public_key_path(), rsa_public) {
+            Err(AetherError::FileWrite(err))
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Load an identity from the default location on the filesystem
+    /// Reads the private key from the default location
+    pub fn load() -> Result<Id, AetherError> {
+        let private_pem = match fs::read(Self::get_private_key_path()) {
+            Ok(data) => data,
+            Err(err) => return Err(AetherError::FileRead(err)),
+        };
+
+        let rsa = Rsa::private_key_from_pem(&private_pem)?;
+
+        Ok(Id { rsa })
+    }
+
+    /// Try to load the identity from the default location on the filesystem or create a new
+    /// identity. If a new identity is created, it is stored in the default location
+    pub fn load_or_generate() -> Result<Id, AetherError> {
+        match Self::load() {
+            Ok(id) => Ok(id),
+            Err(AetherError::FileRead(err)) => {
+                println!("Error reading key: {}", err);
```

Comment:
This can be just a warning or info, error isn't required here imo. Wouldn't it be quite natural for it not to be present on first time execution?

Reply:
Ok will make the change
However can we do this in a follow up PR replacing all `println!` messages with the correct `log` based replacements?
I haven't implemented logging for tests yet. So, if I use `log` functions, I cannot see the messages. I tried earlier but couldn't get it working so thought of doing it later.

Reply:
Cool cool, just put a comment beside it so that we remember